source/ptreferential/tests/ptret_test.cpp

```diff
@@ -428,3 +429,91 @@ BOOST_AUTO_TEST_CASE(find_path_test){
     // Type qui n'existe pas dans le graph : il n'y a pas de chemin
     BOOST_CHECK_THROW(find_path(navitia::type::Type_e::Unknown), ptref_error);
 }
+
+//helper to get default values
+std::vector<nt::idx_t> query(nt::Type_e requested_type, std::string request,
+                             const nt::Data& data,
+                             boost::optional<boost::posix_time::ptime> since = boost::none,
+                             boost::optional<boost::posix_time::ptime> until = boost::none,
+                             bool fail = false) {
+    try {
+       return navitia::ptref::make_query(requested_type, request, {}, navitia::type::OdtLevel_e::all, since, until, data);
+    } catch (const navitia::ptref::ptref_error& e) {
+        if (fail) {
+            throw e;
+        }
+        BOOST_CHECK_MESSAGE(false, " error on ptref request" + std::string(e.what()));
+        return {};
+    }
+}
+
+/*
+ * Test the filtering on the period
+  */
+BOOST_AUTO_TEST_CASE(vj_filtering) {
+    ed::builder builder("20130311");
+    builder.generate_dummy_basis();
+    // Date  11    12    13    14
+    // A      -   08:00 08:00   -
+    // B    10:00 10:00   -   10:00
+    // C      -     -     -   10:00
+    builder.vj("A", "0110")("stop1", "08:00"_t)("stop2", "09:00"_t);
+    builder.vj("B", "1011")("stop3", "10:00"_t)("stop2", "11:00"_t);
+    builder.vj("C", "1000")("stop3", "10:00"_t)("stop2", "11:00"_t);
+    builder.finish();
+    nt::idx_t a = 0;
+    nt::idx_t b = 1;
+    nt::idx_t c = 2;
+
+    auto check = [](const std::vector<nt::idx_t>& col, std::initializer_list<nt::idx_t> ref) {
```

Comment:
`std::initializer_list<nt::idx_t> ref` -> can't you use `const std::set<nt::idx_t>& ref` directly?